### ISSUE TITLE
Why don't you replace `var` with `const` and `let`?

### ISSUE THREAD

**Issue by CreativeGP** [axios/axios#1509]
#### Summary

It seems that there are many `var` in the source code.
And using `const` and `let` instead is [more preferred](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75).
So, why don't you replace `var` with `const` and `let`?
Tell me the reason that you make `var` be left?

**Comment by r3wt** [comment]
@CreativeGP it would just be compiled back to `var` anyway for compatibility reasons.

**Comment by chinesedfan** [comment]
> another common case is variables holding a primitive value that need to be "globally" scoped but modifiable from any child scope. you can't use const or let for that.

@r3wt Sorry, I still don't very understand. Usually, child scope can visit variables in the parent context. Or can you simplify your example codes to show the point?

```
(function() {
    let a = 1;
    (function() {
        console.log(a); // 1
    })();
})();
```

> additionally, you can't declare const or let inside of a switch statement.

Consider following codes, is there any problem? I ran in Node.js 8.9.0.

```
(function(n) {
    switch (n) {
    case 1:
        let x = 2;
        const y = 3;
        console.log(x, y); // 2 3
        break;
    }
})(1);
```

**Comment by CreativeGP** [comment]
@r3wt Thank you for answering.
I see. There are certainly some reason to use `var` even now.
I've wandered in search for that.

**Comment by r3wt** [comment]
@chinesedfan no, `let` is block scoped, so its only available in its scope and any child scopes, unlike `var`.

```js
var age = 100;
if(age > 12) {
  let dogYears = age * 7;
  console.log(`You are ${dogYears} dog years old!`);
}
console.log(dogYears); // error because it's scoped only to the above block
```
const is also block/function scoped, but it is hoisted. therefore declaring it inside of a switch is undefined behavior. in your example, add another `case` in your `switch` statement with the same `const` being declared, and it will crash with an error.

**Comment by chinesedfan** [comment]
@r3wt Thanks for your explanations. I know differences between `var` and `let/const`, especially for scope hoisting. Forgive me to discuss a little more in this issue.

> const is also block/function scoped, but it is hoisted.

I don't think `const` is hoisted. Only `var` can be hoisted, which is the biggest difference with `let/const`. In your previous example, replacing `let` with `const` will result in an error, too.

> add another case in your switch statement with the same const being declared, and it will crash with an error.

That's right. Wrapping each case with a block is a workaround.

> https://github.com/r3wt/express-map/blob/master/index.js#L17

For this example, replacing `var` with `let` should still be able to work. Please correct me if I missed something. :)

In summary, since ES6 provides `let/const`, we should always try to use them. The `var` scope hoisting is easy to be confusing.

**Comment by CreativeGP** [comment]
There seems to be no respond for the above comment?
It passed a month after last comment, and I don't want this issue be open and silent.
So I'll set this closed.
